### PR TITLE
Physics for equippables

### PR DESCRIPTION
**Pull request by hamzzam**
## Summary

Implement drop functionality for equippables.
Handle equippables network ownership transfer logic.
Integrate physics with equippables/dynamic objects.
Fix some networking bugs in EquippableSystem.
Fix physics shape buggy scale when custom scale of object is used in editor.

## Checklist
- [ ] Pre-push checks pass `npm run check`
  - [ ] Linter passing via `npm run lint`
  - [ ] Unit & Integration tests passing via `npm run test:packages`
  - [ ] Docker build process passing via `npm run build-client`
- [ ] If this PR is still a WIP, convert to a draft 
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References

References to pertaining issue(s)

## QA Steps

1. `git checkout physics-for-equippables`
2. `npm install`
3. `npm run dev-reinit`
4. `npm run dev`

To test the equippables do the following:

1. In the editor, add a model node and set its interaction type to Equippable, update interaction distance to something suitable e.g. 5 and then save.
2. Then load the location, take the avatar near the equippable model and press E to pick the object. Load up multiple clients to test the multiplayer behaviour.
3. Move around with the picked object, press U to drop the object anywhere.
4. Repeat steps 2 & 3 on the clients any number of times.

## Reviewers

@HexaField @speigg @NateTheGreatt

### CODE REVIEW

File: packages/engine/src/physics/functions/createCollider.ts [hunk 1]
```diff
@@ -74,8 +76,16 @@ export const createShape = (entity: Entity, mesh: Mesh, shapeOptions: ShapeOptio
   // type is required
   if (!shapeOptions.type) return undefined!
 
-  const scale = mesh.scale ?? new Vector3(1, 1, 1)
+  // Get accurate scale relative to all parents
+  let scale = mesh.scale ?? new Vector3(1, 1, 1)
+  let currentChild = mesh as any
+  while (currentChild && currentChild.uuid !== Engine.scene.uuid) {
```
Comment (HexaField) [inline packages/engine/src/physics/functions/createCollider.ts:82]: we want to use object3d.traverseParents to apply this, and we may be able to update world matricies and use getWorldScale here

File: packages/engine/src/interaction/systems/EquippableSystem.ts [hunk 2]
```diff
@@ -51,56 +60,52 @@ export default async function EquippableSystem(world: World): Promise<System> {
...
-      // const equippedTransform = getComponent(equippedEntity, TransformComponent)
-      // const collider = getComponent(equippedEntity, ColliderComponent)
-      // if (collider) {
-      //   // collider.body.type = BodyType.DYNAMIC
-      //   teleportRigidbody(collider.body, equippedTransform.position, equippedTransform.rotation)
-      // }
+      const equippedTransform = getComponent(equippedEntity, TransformComponent)
+      const collider = getComponent(equippedEntity, ColliderComponent)
+      if (collider) {
+        let phsyxRigidbody = collider.body as PhysX.PxRigidBody
```
Comment (HexaField) [inline packages/engine/src/interaction/systems/EquippableSystem.ts:101]: lets create some `changeRigidbodyType(body: PhysX.PxRigidBody, type: BodyType)` in `Physics.ts`  to abstract this into unified calls

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/physics/functions/createCollider.ts (hunk 1 above)
Comment (hamzzam) [reply packages/engine/src/physics/functions/createCollider.ts]: Why didn't I think of that the first time😄.
Updated now.

File: packages/engine/src/interaction/systems/EquippableSystem.ts (hunk 2 above)
Comment (hamzzam) [reply packages/engine/src/interaction/systems/EquippableSystem.ts:98]: Added.

File: packages/engine/src/interaction/systems/InteractiveSystem.ts [hunk 3]
```diff
@@ -48,10 +48,11 @@ export default async function InteractiveSystem(world: World): Promise<System> {
 
   return () => {
     for (const entity of interactiveQuery.enter(world)) {
+      const interactoinData = getComponent(entity, InteractableComponent).data
```
Comment (HexaField) [inline packages/engine/src/interaction/systems/InteractiveSystem.ts:51]: typo `interactoinData` should be `interactionData`

**Review by HexaField — CHANGES_REQUESTED** [review]

**Review by HexaField — APPROVED** [review]

**Pull request merged**